src/dataflow/src/render/mod.rs

```diff
@@ -841,25 +841,25 @@ where
                         .collect::<Vec<_>>();
 
                     let (ok_collection, err_collection) = self.collection(input).unwrap();
-                    let (ok_collection, new_err_collection) =
-                        ok_collection.flat_map_fallible({
-                            let mut row_packer = repr::RowPacker::new();
-                            move |input_row| {
-                                let datums = input_row.unpack();
-                                let replace = replace.clone();
-                                let temp_storage = RowArena::new();
-                                let exprs = exprs
-                                    .iter()
-                                    .map(|e| e.eval(&datums, &temp_storage))
-                                    .collect::<Result<Vec<_>, _>>();
-                                let exprs = match exprs {
-                                    Ok(exprs) => exprs,
-                                    Err(e) => return vec![Err(e.into())],
-                                };
-                                let output_rows = func.eval(exprs, &temp_storage);
-                                output_rows
-                                    .into_iter()
-                                    .map(|output_row| {
+                    let (ok_collection, new_err_collection) = ok_collection.explode_fallible({
+                        let mut row_packer = repr::RowPacker::new();
+                        move |input_row| {
+                            let datums = input_row.unpack();
+                            let replace = replace.clone();
```

Comment:
Not actually a change you are making, but note to self / discussion: do we really need to clone this?